godot-cell/src/lib.rs

```diff
@@ -69,19 +77,62 @@ impl<T> GdCell<T> {
     /// Returns a new shared reference to the contents of the cell.
     ///
     /// Fails if an accessible mutable reference exists.
+    ///
+    /// Blocks if an other thread currently holds a mutable reference.
     pub fn borrow(self: Pin<&Self>) -> Result<RefGuard<'_, T>, Box<dyn Error>> {
+        #[cfg(feature = "threads")]
+        let read_lock = 'lock: {
+            let mut_thread_id = self.get_ref().mut_thread.read_recursive();
+            let shared_count = self.state.lock().unwrap().borrow_state.shared_count();
+
+            // If there is already a shared reference, we don't need to care about the thread id.
+            if shared_count > 0 {
+                break 'lock mut_thread_id;
+            }
+
+            self.get_ref().claim_mut_thread(mut_thread_id)
```

Comment:
No `break` and label needed, or am I missing something?
```suggestion
            // If there is already a shared reference, we don't need to care about the thread ID.
            if shared_count > 0 {
                mut_thread_id
            } else {
                self.get_ref().claim_mut_thread(mut_thread_id)
            }
```

Reply:
Yes, after I factored out `claim_mut_thread`, this makes sense now.